sources/ec2/az.go

```diff
@@ -0,0 +1,85 @@
+package ec2
+
+import (
+	"context"
+	"errors"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	"github.com/overmindtech/aws-source/sources"
+	"github.com/overmindtech/sdp-go"
+)
+
+// AvailabilityZoneInputMapper Maps source calls to the correct input for the AZ API
+func AvailabilityZoneInputMapper(scope, query string, method sdp.RequestMethod) (*ec2.DescribeAvailabilityZonesInput, error) {
+	var input ec2.DescribeAvailabilityZonesInput
+
+	switch method {
+	case sdp.RequestMethod_GET:
+		input = ec2.DescribeAvailabilityZonesInput{
+			ZoneNames: []string{
+				query,
+			},
+		}
+	case sdp.RequestMethod_LIST:
+		input = ec2.DescribeAvailabilityZonesInput{}
+	default:
+		return nil, errors.New("unsupported method")
+	}
```

Comment:
This feels like it'll become repetitive very quickly. maybe have an `AvailabilityZone{Get,List,Search}InputMapper` instead?